src/devices/Hx711I2c/Hx711I2c.cs

```diff
@@ -0,0 +1,349 @@
+            // It does not seem to be possible to read from this register
+            set => WriteUInt16(Hx711I2cRegister.REG_SET_CAL_THRESHOLD, value);
+        }
+
+        /// <summary>
+        /// Hx711I2c - DFRobot KIT0176: I2C 1kg Weight Sensor Kit
+        /// </summary>
+        public Hx711I2c(I2cDevice i2cDevice)
+        {
+            _i2c = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            Offset = DefaultOffset;
+            CalibrationScale = GetAutomaticCalibrationScale();
+        }
+
+        /// <summary>
+        /// Gets I2C address depending on A0 and A1 pin settings.
+        /// </summary>
+        /// <param name="a0">Value of A0 pin.</param>
+        /// <param name="a1">Value of A1 pin.</param>
+        /// <returns>Address of the device.</returns>
+        public static int GetI2cAddress(PinValue a0, PinValue a1)
+            => DefaultI2cAddress + ((bool)a0 ? 1 : 0) + ((bool)a1 ? 2 : 0);
+
+        /// <summary>
+        /// Read current weight and use it as 0. Equivalent to pressing RST button.
+        /// </summary>
+        [Command]
+        public void Tare()
+        {
+            Offset = GetRawReading();
+        }
+
+        /// <summary>
+        /// Re-initializes the sensor and sets arbitrarly chosen calibration values.
+        /// </summary>
+        [Command]
+        public void ResetSensor()
+        {
+            // This is purposefully not called "Reset" to not confuse with RST button and Tare functionality
+            const byte REG_CLEAR_REG_STATE = 0x65;
+            WriteByte(Hx711I2cRegister.REG_DATA_INIT_SENSOR, REG_CLEAR_REG_STATE);
+
+            AutomaticCalibrationWeight = 100;
+            AutomaticCalibrationThreshold = 10;
+
+            Offset = DefaultOffset;
+            CalibrationScale = GetAutomaticCalibrationScale();
+        }
+
+        /// <summary>
+        /// Gets data from sensor with scale which was determined with automatic calibration made when CAL button was pressed.
+        /// This value might differ from actual calibration scale used if no measurements have been made after CAL button was pressed.
+        /// </summary>
+        /// <returns>Calibration scale</returns>
+        public float GetAutomaticCalibrationScale()
+        {
+            // TODO: in netstandard2.0 we cannot use ReadOnlySpan overload of BitConverter
+            // TODO: once we drop support we should use Span overload
+            // TODO: once we're at 6.0 or higher we should use BinaryPrimitives.ReadSingleBigEndian
+            byte[] calibrationBytes = new byte[4];
+            ReadRegister(Hx711I2cRegister.REG_DATA_GET_CALIBRATION, calibrationBytes);
+            Reverse4ByteArray(calibrationBytes);
+            return BitConverter.ToSingle(calibrationBytes, 0);
+        }
+
+        /// <summary>
+        /// Same as Tare but will also blink LED next to RST button.
+        /// </summary>
+        [Command]
+        public void HardwareTare()
```

Comment:
NIT: Should we just have one Tare button that takes in a bool for blinking the LED?